pages/my-account/order-history.vue

```diff
@@ -144,10 +162,46 @@ export default defineComponent({
       }
     }
 
+    const selectFilter = (filterValue) => {
+      const currentIndex = filters.value.indexOf(filterValue)
+      if (currentIndex > -1) {
+        filters.value.splice(currentIndex, 1)
+        facetAllOptions.value.find((facet) => facet.filterValue === filterValue).isApplied = false
+      } else {
+        filters.value.push(filterValue)
+        facetAllOptions.value.find((facet) => facet.filterValue === filterValue).isApplied = true
+      }
+    }
+
     useAsync(async () => {
-      await getOrders({ filters: "" })
+      const facetsFromURL = getFacetsFromURL()
+      facetsFromURL.filters.forEach((filter) => {
+        facetAllOptions.value.find((facet) => facet.filterValue === filter).isApplied = true
+      })
+
+      facetAllOptions.value.forEach((facet) => {
```

Comment:
it looks like you're duplicating the state in multiple spots.
Theres a "filter" ref, and you're mutating the "facetAllOptions" and getting pulling values from the url?
Setup one approach and stick with it

Reply:
Resolved